### PR TITLE
[New Rule] Potential OpenSSH Backdoor Logging Activity

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/748

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/linux/credential_access_ssh_backdoor_log.toml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+creation_date = "2020/12/21"
+maturity = "production"
+updated_date = "2020/12/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies SSH client or server process creating or writing to a known SSH Backdoor Log filename. Adversaries may modify
+SSH related binaries for persistence or credential access via patching sensitive functions to enable unauthorized access
+or log SSH credentials for exfiltration.
```
Comment (peasead) [inline rules/linux/credential_access_ssh_backdoor_log.toml:11]:
```suggestion
This rule identifies an SSH client or server process creating, or writing to, a known SSH backdoor log file.
```

File: rules/linux/credential_access_ssh_backdoor_log.toml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+"""
+false_positives = ["Trusted SSH Executables Updates."]
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "lucene"
+license = "Elastic License"
+name = "Potential OpenSSH Backdoor Logging Activity"
```
Comment (peasead) [inline rules/linux/credential_access_ssh_backdoor_log.toml:18]:
```suggestion
name = "Potential OpenSSH Backdoor Logging Activity"
note = "Adversaries may modify SSH related binaries for persistence or credential access via patching sensitive functions to enable unauthorized access or to log SSH credentials for exfiltration."
```

File: rules/linux/credential_access_ssh_backdoor_log.toml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+updated_date = "2020/12/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies SSH client or server process creating or writing to a known SSH Backdoor Log filename. Adversaries may modify
+SSH related binaries for persistence or credential access via patching sensitive functions to enable unauthorized access
+or log SSH credentials for exfiltration.
+"""
+false_positives = ["Trusted SSH Executables Updates."]
```
Comment (peasead) [inline rules/linux/credential_access_ssh_backdoor_log.toml:13]:
```suggestion
false_positives = ["Trusted SSH executable updates."]
```

File: rules/linux/credential_access_ssh_backdoor_log.toml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "lucene"
+license = "Elastic License"
+name = "Potential OpenSSH Backdoor Logging Activity"
+references = ["https://github.com/eset/malware-ioc/tree/master/sshdoor"]
+risk_score = 73
+rule_id = "f28e2be4-6eca-4349-bdd9-381573730c22"
+severity = "high"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Persistence", "Credential Access"]
+type = "query"
```
Comment (peasead) [inline rules/linux/credential_access_ssh_backdoor_log.toml:24]:
```suggestion
type = "eql"
```

File: rules/linux/credential_access_ssh_backdoor_log.toml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+author = ["Elastic"]
+description = """
+Identifies SSH client or server process creating or writing to a known SSH Backdoor Log filename. Adversaries may modify
+SSH related binaries for persistence or credential access via patching sensitive functions to enable unauthorized access
+or log SSH credentials for exfiltration.
+"""
+false_positives = ["Trusted SSH Executables Updates."]
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "lucene"
```
Comment (peasead) [inline rules/linux/credential_access_ssh_backdoor_log.toml:16]:
```suggestion
language = "eql"
```

**Review by peasead — APPROVED** [review]
LGTM

File: rules/linux/credential_access_ssh_backdoor_log.toml [hunk 6]
```diff
@@ -0,0 +1,61 @@
+[metadata]
+creation_date = "2020/12/21"
+maturity = "production"
+updated_date = "2020/12/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule identifies an SSH client or server process creating, or writing to, a known SSH backdoor log file.
```
Comment (bm11100) [inline rules/linux/credential_access_ssh_backdoor_log.toml:9]:
```suggestion
Identifies a Secure Shell (SSH) client or server process creating or writing to a known SSH backdoor log file. Adversaries may modify SSH related binaries for persistence or credential access via patching sensitive functions to enable unauthorized access or to log SSH credentials for exfiltration.
```

Comment (bm11100) [reply rules/linux/credential_access_ssh_backdoor_log.toml:9]: may need to re lint

File: rules/linux/credential_access_ssh_backdoor_log.toml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+"""
+false_positives = ["Updates to approved and trusted SSH executables can trigger this rule."]
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Potential OpenSSH Backdoor Logging Activity"
+note = "Adversaries may modify SSH related binaries for persistence or credential access via patching sensitive functions to enable unauthorized access or to log SSH credentials for exfiltration."
```
Comment (bm11100) [inline rules/linux/credential_access_ssh_backdoor_log.toml:17]:
```suggestion

```
belongs in description.

File: rules/linux/credential_access_ssh_backdoor_log.toml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+query = '''
+file where event.type == "change" and process.executable in ("/usr/sbin/sshd", "/usr/bin/ssh") and
+ (
+   wildcard(file.name, ".*", "~*") or
+   file.extension in ("in", "out", "ini", "h", "gz", "so", "sock", "sync", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9") or
+   wildcard(file.path, "/private/etc/*--", "/usr/share/*", "/usr/include/*", "/usr/local/include/*", "/private/tmp/*", "/private/var/tmp/*",
+                       "/usr/tmp/*", "/usr/share/man/*", "/usr/local/share/*", "/usr/lib/*.so.*", "/private/etc/ssh/.sshd_auth",
+                       "/usr/bin/ssd", "/private/var/opt/power", "/private/etc/ssh/ssh_known_hosts", "/private/var/html/lol", "/private/var/log/utmp", "/private/var/lib"
+	       )
+  )
```
Comment (bm11100) [inline rules/linux/credential_access_ssh_backdoor_log.toml:34]:
```suggestion
file where event.type == "change" and process.executable in ("/usr/sbin/sshd", "/usr/bin/ssh") and
 (
   file.name:(".*", "~*") or
   file.extension:("in", "out", "ini", "h", "gz", "so", "sock", "sync", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9") or
   file.path:(
     "/private/etc/*--", 
     "/usr/share/*", 
     "/usr/include/*", 
     "/usr/local/include/*", 
     "/private/tmp/*", 
     "/private/var/tmp/*",
     "/usr/tmp/*", 
     "/usr/share/man/*", 
     "/usr/local/share/*", 
     "/usr/lib/*.so.*", 
     "/private/etc/ssh/.sshd_auth",
     "/usr/bin/ssd", 
     "/private/var/opt/power", 
     "/private/etc/ssh/ssh_known_hosts", 
     "/private/var/html/lol", 
     "/private/var/log/utmp", 
     "/private/var/lib"
     )
  )
```

Comment (bm11100) [reply rules/linux/credential_access_ssh_backdoor_log.toml:34]: removed the case sensitive `wildcard` function and made it a bit more readable. should look like below - 
```
file where event.type == "change" and process.executable in ("/usr/sbin/sshd", "/usr/bin/ssh") and
 (
   file.name:(".*", "~*") or
   file.extension:("in", "out", "ini", "h", "gz", "so", "sock", "sync", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9") or
   file.path:(
     "/private/etc/*--", 
     "/usr/share/*", 
     "/usr/include/*", 
     "/usr/local/include/*", 
     "/private/tmp/*", 
     "/private/var/tmp/*",
     "/usr/tmp/*", 
     "/usr/share/man/*", 
     "/usr/local/share/*", 
     "/usr/lib/*.so.*", 
     "/private/etc/ssh/.sshd_auth",
     "/usr/bin/ssd", 
     "/private/var/opt/power", 
     "/private/etc/ssh/ssh_known_hosts", 
     "/private/var/html/lol", 
     "/private/var/log/utmp", 
     "/private/var/lib"
     )
  )
```

File: rules/linux/credential_access_ssh_backdoor_log.toml [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+severity = "high"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Persistence", "Credential Access"]
+type = "eql"
+
+query = '''
+file where event.type == "change" and process.executable in ("/usr/sbin/sshd", "/usr/bin/ssh") and
+ (
+   file.name:(".*", "~*") or
+   file.extension:("in", "out", "ini", "h", "gz", "so", "sock", "sync", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9") or
+   file.path:(
```
Comment (brokensound77) [inline rules/linux/credential_access_ssh_backdoor_log.toml:34]:
```suggestion
   file.name : (".*", "~*") or
   file.extension in ("in", "out", "ini", "h", "gz", "so", "sock", "sync", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9") or
   file.path : (
```

File: rules/linux/credential_access_ssh_backdoor_log.toml [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+     "/private/etc/ssh/ssh_known_hosts", 
+     "/private/var/html/lol", 
+     "/private/var/log/utmp", 
+     "/private/var/lib",
+     "/var/run/sshd/sshd.pid",
+     "/var/run/nscd/ns.pid",
+     "/var/run/udev/ud.pid",
+     "/var/run/udevd.pid"
+     )
+  )
```
Comment (bm11100) [inline rules/linux/credential_access_ssh_backdoor_log.toml:57]:
```suggestion
file where event.type == "change" and process.executable : ("/usr/sbin/sshd", "/usr/bin/ssh") and
  (
    file.name : (".*", "~*") or
    file.extension : ("in", "out", "ini", "h", "gz", "so", "sock", "sync", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9") or
    file.path : 
    (
      "/private/etc/*--", 
      "/usr/share/*", 
      "/usr/include/*", 
      "/usr/local/include/*", 
      "/private/tmp/*", 
      "/private/var/tmp/*",
      "/usr/tmp/*", 
      "/usr/share/man/*", 
      "/usr/local/share/*", 
      "/usr/lib/*.so.*", 
      "/private/etc/ssh/.sshd_auth",
      "/usr/bin/ssd", 
      "/private/var/opt/power", 
      "/private/etc/ssh/ssh_known_hosts", 
      "/private/var/html/lol", 
      "/private/var/log/utmp", 
      "/private/var/lib",
      "/var/run/sshd/sshd.pid",
      "/var/run/nscd/ns.pid",
      "/var/run/udev/ud.pid",
      "/var/run/udevd.pid"
    )
  )
```

Comment (bm11100) [reply rules/linux/credential_access_ssh_backdoor_log.toml:57]: otherwise, LGTM

**Review by bm11100 — APPROVED** [review]

**Pull request merged**